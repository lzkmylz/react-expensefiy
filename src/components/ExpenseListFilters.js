import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'
import uuid from 'uuid';

export class ExpenseListFilters extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            calendarFocused: null
        };
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    };

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };

    onSortChange = (e) => {
        let sortBy = e.target.value;
        if(sortBy === 'date') {
            this.props.sortByDate();
        } else if(sortBy === 'amount') {
            this.props.sortByAmount();
        }
    };
    

    render() {
        return (
            <div className="content_container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input 
                        className="text-input" 
                        type="text" 
                        value={this.props.filters.text} 
                        onChange={this.onTextChange} 
                        placeholder="Search Expenses"
                        />
                    </div>
                    <div className="input-group__item">
                        <select className="select" value={this.props.filters.sortBy} onChange={this.onSortChange} >
                        <option value="date" >Date</option>
                        <option value="amount" >Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker 
                        startDate={this.props.filters.startDate}
                        startDateId={uuid()}
                        endDate={this.props.filters.endDate}
                        endDateId={uuid()}
                        onDatesChange={this.onDatesChange}
                        focusedInput={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        showClearDates={true}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);