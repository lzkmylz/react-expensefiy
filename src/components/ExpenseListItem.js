import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <p>description: {description}</p>
        </Link>
        <p>amount: {numeral(amount).format('$0,0.00')}</p>
        <p>createdAt: {moment(createdAt).format('YYYY-MM-DD')}</p>
    </div>
);

export default ExpenseListItem;