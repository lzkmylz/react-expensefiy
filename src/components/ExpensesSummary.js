import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expense';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <p>expenseCount: {props.expenses.length}</p>
        <p>expensesTotal: {numeral(selectExpensesTotal(props.expenses)).format('$0,0.00')}</p>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);