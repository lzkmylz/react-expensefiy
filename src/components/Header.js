import React from 'react';
import { NavLink } from 'react-router-dom';
import { startLogOut } from '../actions/auth';
import { connect } from 'react-redux';


export const Header = ({ startLogOut }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName="is-active" > Home </NavLink>
        <NavLink to="/create" activeClassName="is-active" > Create </NavLink>
        <NavLink to="/edit" activeClassName="is-active" > Edit </NavLink>
        <NavLink to="/help" activeClassName="is-active" > Help </NavLink>
        <button onClick={startLogOut} >LogOut</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => {
        dispatch(startLogOut());
    }
});

export default connect(undefined, mapDispatchToProps)(Header);