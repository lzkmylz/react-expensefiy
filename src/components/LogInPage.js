import React from 'react';
import { connect } from 'react-redux';
import { startLogIn } from '../actions/auth';

export const LogInPage = ({ startLogIn }) => (
    <div className="box-layout" >
        <div className="box-layout__box" >
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className="button" onClick={startLogIn} >Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogIn: () => dispatch(startLogIn())
});

export default connect(undefined, mapDispatchToProps)(LogInPage);