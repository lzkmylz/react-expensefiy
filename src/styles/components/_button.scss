.button {
    background: $blue;
    color: white;
    display: inline block;
    text-decoration: none;
    border: none;
    line-height: 1;
    font-size: $font-size-large;
    padding: $s-size;
    font-weight: 300;
}

.button--link {
    background: none;
    color: white;
    border: none;
}