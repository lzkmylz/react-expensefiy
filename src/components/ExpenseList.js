import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expense';

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobil">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        <h1>ExpenseList</h1>
        {
            props.expenses.length === 0 ? (<p>No Expense Found!</p>) : 
            (props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense} />
            }))
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);