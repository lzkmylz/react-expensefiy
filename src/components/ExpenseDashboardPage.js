import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
        <ExpenseSummary />
    </div>
);

export default ExpenseDashboardPage;