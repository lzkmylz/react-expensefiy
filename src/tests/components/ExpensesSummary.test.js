import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { ExpensesSummary } from '../../components/ExpensesSummary';

const expenses = [
    {
        id: 1,
        description: 'Gum',
        note: '',
        amount: 195,
        createdAt: 0
    },{
        id: 2,
        description: 'Rent',
        note: '',
        amount: 109500,
        createdAt: moment(0).subtract(4, 'days').valueOf()
    }
];

test('should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with 2 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});