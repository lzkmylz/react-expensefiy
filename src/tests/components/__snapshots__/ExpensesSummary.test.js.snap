// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenses={
      Array [
        Object {
          "amount": 195,
          "createdAt": 0,
          "description": "Gum",
          "id": 1,
          "note": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          expenseCount: 
          1
        </p>,
        <p>
          expensesTotal: 
          $195.00
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "expenseCount: ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "expenseCount: ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "expensesTotal: ",
            "$195.00",
          ],
        },
        "ref": null,
        "rendered": Array [
          "expensesTotal: ",
          "$195.00",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            expenseCount: 
            1
          </p>,
          <p>
            expensesTotal: 
            $195.00
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "expenseCount: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "expenseCount: ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "expensesTotal: ",
              "$195.00",
            ],
          },
          "ref": null,
          "rendered": Array [
            "expensesTotal: ",
            "$195.00",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`should render ExpensesSummary with 2 expenses 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenses={
      Array [
        Object {
          "amount": 195,
          "createdAt": 0,
          "description": "Gum",
          "id": 1,
          "note": "",
        },
        Object {
          "amount": 109500,
          "createdAt": -345600000,
          "description": "Rent",
          "id": 2,
          "note": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          expenseCount: 
          2
        </p>,
        <p>
          expensesTotal: 
          $109,695.00
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "expenseCount: ",
            2,
          ],
        },
        "ref": null,
        "rendered": Array [
          "expenseCount: ",
          2,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "expensesTotal: ",
            "$109,695.00",
          ],
        },
        "ref": null,
        "rendered": Array [
          "expensesTotal: ",
          "$109,695.00",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            expenseCount: 
            2
          </p>,
          <p>
            expensesTotal: 
            $109,695.00
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "expenseCount: ",
              2,
            ],
          },
          "ref": null,
          "rendered": Array [
            "expenseCount: ",
            2,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "expensesTotal: ",
              "$109,695.00",
            ],
          },
          "ref": null,
          "rendered": Array [
            "expensesTotal: ",
            "$109,695.00",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
